grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java

```diff
@@ -416,6 +427,46 @@ private static Descriptor findTypeDescriptor(FileDescriptor file, FieldDescripto
         return file.findMessageTypeByName(messageType.getName());
     }
 
+    // to make it more efficient, we calculate whether extract response body one time
+    private static boolean needExtractResponseBody(List<FieldDescriptor> topLevelFields, String responseBody) {
+        if (StringUtil.isNullOrEmpty(responseBody)) {
+            return false;
+        }
+        for (FieldDescriptor fieldDescriptor: topLevelFields) {
+            if (fieldDescriptor.getName().equals(responseBody)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Nullable
+    private static Function<HttpData, HttpData> generateResponseBodyConverter(TranscodingSpec spec) {
+        final String responseBody = spec.httpRule.getResponseBody();
+        if (!spec.needExtractResponseBody) {
+            return null;
+        } else {
+            return httpData -> {
+                try (HttpData body = httpData) {
+                    final byte[] array = body.array();
+                    final ObjectMapper om = new ObjectMapper();
+                    final ObjectReader reader = om.reader();
+                    try {
+                        final JsonNode jsonNode = reader.readTree(array);
+                        final String responseBodyJsonKey =
+                                CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, responseBody);
+                        final JsonNode responseBodyJsonNode = jsonNode.get(responseBodyJsonKey);
+                        final ObjectWriter writer = om.writer();
+                        final byte[] bytes = writer.writeValueAsBytes(responseBodyJsonNode);
```

Comment:
Don't need to create `ObjectWrite`. Should we directly write the value via `JacksonUtil.writeValueAsBytes()`?